### ISSUE TITLE
feature request: make `--dist worksteal` the default

### ISSUE THREAD

**Issue by charles-cooper** [pytest-dev/pytest-xdist#1042]
i have been finding that `--dist worksteal` is generally more performant than `--dist load` -- sometimes substantially. ideally, `--dist worksteal` should be the default, since it's simply a better algorithm. this would also address https://github.com/pytest-dev/pytest-xdist/issues/890 by making the other schedulers (loadfile, loadgroup etc) include work stealing.

**Comment by amezin** [comment]
No, it won't magically make file/group/scope schedulers do work-stealing based scheduling

**Comment by charles-cooper** [comment]
> No, it won't magically make file/group/scope schedulers do work-stealing based scheduling

ah, right, my suggestion was that the other schedulers could use workstealing instead of what they are doing right now. i did not mean to imply that it would happen magically!
